core/services/log/registrations.go

```diff
@@ -89,7 +89,10 @@ func (r *registrations) removeSubscriber(reg registration) (needsResubscribe boo
 	}
 
 	needsResubscribe = l.removeSubscriber(reg)
-	r.resetHighestNumConfirmations()
+
+	if reg.opts.NumConfirmations >= r.highestNumConfirmations {
```

Comment:
Why this check?

Reply:
could have used == instead of >=. But it's because we only need to update this variable if removed NumConfirmations is on the boundary

Reply:
added a comment as well